src/url.rs

```diff
@@ -177,15 +177,25 @@ impl PyUrl {
         };
         let mut url = format!("{scheme}://{url_host}");
         if let Some(path) = path {
-            url.push('/');
-            url.push_str(path);
+            if !url.ends_with('/') {
+                url.push('/');
+            }
+            if path.starts_with('/') {
+                url.push_str(path.trim_start_matches('/'));
+            } else {
+                url.push_str(path);
+            }
```

Comment:
I think trimming the path is too aggressive, because `///foo` as a path is technically meaningful and 

```suggestion
            if !path.starts_with('/') {
                url.push('/');
            }
```